canonical-web-and-design/maas-ui
# Add pager component and paginate users table.

## Done
1. Create `Pager` component.
2. Added pagination to the User table.

The `Pager` component supports both truncated and simple pagination [as described in the vanilla docs](https://docs.vanillaframework.io/patterns/pagination/).

![image](https://user-images.githubusercontent.com/130286/62511737-2b5b6e80-b869-11e9-904d-c6edeb47e8ab.png)

## QA
1. Ensure the users table renders (if you have more than 12 users, the pager should render).
2. Ensure the *previous* button is disabled when on the first item.
3. Ensure the *next* button is disabled when on the last item.
4. Click page 1-3, the pager should not truncate initially e.g. 
![image](https://user-images.githubusercontent.com/130286/62511758-43cb8900-b869-11e9-8022-db0af9d2f879.png)

but should truncate past 4.
5. Click the last page, or 1-3 pages from the last, it should not truncate initially, .e.g. 
![image](https://user-images.githubusercontent.com/130286/62511590-acfecc80-b868-11e9-8725-485bba26c23f.png)

6. In `Users.js` manually hack the `Pager` to have fewer `totalItems` than `itemsPerPage`, the `Pager` should not render. (Probably easier than hacking users in your state).
7. Hack the pager to only have a few pages, e.g. `totalItems={50}` and `itemsPerPage={10}`, you should see a simple `Pager` render without truncation.
